packages/textlint-fixer-formatter/src/index.ts

```diff
@@ -46,7 +46,11 @@ ${ex}`);
     };
 }
 
-export function getFormatterList() {
+export interface FormatterDetail {
+    name: string;
+}
+
+export function getFormatterList(): FormatterDetail[] {
```

Comment:
Public API should have explicit interface.

Reply:
@azu, it looks ok for me. But `textlint-formatter` already has `FormatterDetail` interface. To avoid duplication, how about following changes? (the interface is small, though)

```diff
diff --git a/packages/textlint-fixer-formatter/src/index.ts b/packages/textlint-fixer-formatter/src/index.ts
index d8cda30..f0db498 100644
--- a/packages/textlint-fixer-formatter/src/index.ts
+++ b/packages/textlint-fixer-formatter/src/index.ts
@@ -1,6 +1,7 @@
 // LICENSE : MIT
 "use strict";
 import { TextlintFixResult } from "@textlint/kernel";
+import { FormatterDetail } from "textlint-formatter";

 const fs = require("fs");
 const path = require("path");
@@ -46,7 +47,7 @@ ${ex}`);
     };
 }

-export function getFormatterList() {
+export function getFormatterList(): FormatterDetail[] {
     return fs
         .readdirSync(path.join(__dirname, "formatters"))
         .filter((file: string) => {
```

 This will introduce additional dependency to `textlint-formatter`, but since `textlint-fixer-formatter`

Reply:
I think that the duplication is OK.
Because, `textlint-fixer-formatter`'s `FormatterDetail` can be difference with `textlint-formatter`s `FormatterDetail`.
(Also,  `textlint-fixer-formatter` does not depnended on `textlint-formatter` in semantically)

Reply:
I think that rename `FormatterDetail` to `FixerFormatterDetail` is reasone.

@0x6b Do you think about it ?

Reply:
@azu, thank you.

> Because, `textlint-fixer-formatter`'s `FormatterDetail` can be difference with `textlint-formatter`s `FormatterDetail`.

This makes sense for me, and renaming new interface to`FixerFormatterDetail` LGTM.

Reply:
But `packages/textlint/src/options.ts` implementation might be a bit awful...?

Reply:
Not awful as I thought first. Union type is cool.